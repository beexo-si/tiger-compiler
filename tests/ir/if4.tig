/*TIGER_ARGS=-arbol -ir*/
let
    var x:=5
in
    (if x then () else ();
    0)
end
