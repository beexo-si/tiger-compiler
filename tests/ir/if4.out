LetExp{decs=[VarDec{name=x,escape=false,typ=NONE,init=5}],
 body=SeqExp([IfExp{test=VarExp(SimpleVar(x)),then'=UnitExp,else'=UnitExp}0])}
bien!
;;-PROC-L0--:
MOVE(
 TEMP rax,
 ESEQ(
  MOVE(
   TEMP T0,
   CONST 5),
  ESEQ(
   SEQ(
    CJUMP(NE,
     TEMP T0,
     CONST 0,
     L1,L2),
    SEQ(
     LABEL L1,
     SEQ(
      EXP(
       CONST 0),
      SEQ(
       JUMP(        NAME L3),
       SEQ(
        LABEL L2,
        SEQ(
         EXP(
          CONST 0),
         LABEL L3)))))),
   CONST 0)))
;;-END-PROC--:
yes!!
